### ISSUE TITLE
Feature request: Copy-paste over different logisim instances/projects

### ISSUE THREAD

**Issue by BFH-ktt1** [logisim-evolution/logisim-evolution#1050]
Currently logisim allows only copy/paste/duplicate in the same project space (internal clip-board). Kevin Walsh implemented in his holy-cross version a possibility to copy-paste over different projects using the systems clipboard. This feature would be very nice to have. Back-porting of this feature might be more effort as setting it up from scratch.

**Comment by MarcinOrlowski** [comment]
If every element can be serialized for saving, we can pass that representation via clipboard.

**Comment by BFH-ktt1** [comment]
Yes, all can be serialized (as been done in xml in the .circ file) Kevin Walsh used this same format (xml) to do the copy-paste

**Comment by BFH-ktt1** [comment]
What might be more an issue would be a copy in logisim, and for example a paste in a drawing program, than there needs to be an extra writer attached (like the one's used to do picture export in the different formats)

**Comment by MarcinOrlowski** [comment]
We should export SVG. That'd be be the best option.

**Comment by R3dst0ne** [comment]
Is this a duplicate of #56?

**Comment by BFH-ktt1** [comment]
Yes this is correct. Closing #56 as we have a link here

**Comment by hewzhew** [comment]
@BFH-ktt1 I rechecked this against current `main` and the referenced Holy Cross implementation.

The issue still exists for independently launched processes. The layout clipboard is currently a static in-JVM `Clipboard` containing cloned `Component` objects. This already supports multiple projects inside one process, and `SelectionActions.getReplacementMap(...)` can resolve compatible factories in the destination project, but a second process cannot see that object graph.

The Holy Cross branch solved a broader problem with a versioned XML payload, custom system-clipboard flavors, dependency serialization, and dedicated readers for components, circuits, VHDL, and libraries. It remains useful design evidence, but a direct backport would also bring in old clipboard wrappers, UI behavior, and automatic dependency-import machinery that no longer matches current code.

I propose a focused first PR:

- keep the existing in-process clipboard and its undo/redo behavior as the fast path;
- on layout copy/cut, also publish a versioned Logisim-specific system-clipboard flavor whose representation is a UTF-8 XML string;
- include only the selected components/wires plus enough source-library identifiers to resolve their factories;
- on cross-process paste, parse the payload with the current hardened XML setup and reconstruct against libraries/circuits already present in the destination project;
- reuse the current replacement/drop/cancel behavior for missing or name-conflicting factories rather than silently importing dependencies;
- update Paste enablement when the system clipboard flavor changes, and handle a temporarily busy/unavailable clipboard without blocking the EDT;
- add codec tests for wires, built-in attributes, malformed/untrusted XML, unresolved factories, and preservation of the existing same-process path.

This deliberately excludes copying whole circuits/libraries/VHDL models, recursively importing subcircuit dependencies, appearance-editor objects, and SVG/image/plain-text export. Those can be evaluated independently after the component path is stable.

Would this dependency-satisfied, components-and-wires-only first slice be welcome?

**Comment by maehne** [comment]
@hewzhew: Thanks for your analysis! I agree that it makes sense to address this enhancement in multiple steps as outlined by you above. 

Where it makes sense, using/adopting @kevinawalsh's approach (with proper reference) may make sense to limit code divergence to facilitate later back-porting of other enhancements by him. If none of his relevant commits can be adopted/merged, noting these commits as discarded in the PR may help us, too. However, you don't need to spend too much effort on this aspect.
